src/overlays/actors/ovl_En_Daiku/z_en_daiku.c

```diff
@@ -22,38 +51,315 @@ const ActorInit En_Daiku_InitVars = {
+        this->collider.dim.yShift = 0;
+        this->actor.flags |= 0x08000000;
+        if ((gSaveContext.weekEventReg[63] & 0x80) || ((gSaveContext.day == 3) && gSaveContext.isNight)) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    } else if ((gSaveContext.day == 3) && gSaveContext.isNight) {
+        Actor_MarkForDeath(&this->actor);
+    }
+
+    Math_Vec3f_Copy(&this->unk_26C, &this->actor.world.pos);
+    this->unk_280 = this->actor.world.rot.y;
+    this->actor.gravity = -3.0f;
+
+    switch (this->unk_278) {
+        case ENDAIKU_PARAMS_0:
+            this->unk_27E = this->unk_278 * 4 + 4;
+
+        case ENDAIKU_PARAMS_1:
+            SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600A850, &D_06002FA0, this->jointTable, this->morphTable,
+                             17);
+            break;
+
+        case ENDAIKU_PARAMS_2:
+            SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600A850, &D_0600B690, this->jointTable, this->morphTable,
+                             17);
+            break;
+
+        case ENDAIKU_PARAMS_3:
+            SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0600A850, &D_06001114, this->jointTable, this->morphTable,
+                             17);
+            break;
+    }
+
+    func_80943820(this);
+}
+
+void EnDaiku_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDaiku* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void func_8094373C(EnDaiku* this, s32 arg1) {
+    static AnimationHeader* D_809440A4[] = {
+        &D_06002FA0, &D_0600ACD0, &D_0600C92C, &D_06000C44, &D_0600C234,
+        &D_06000600, &D_06001114, &D_0600B690, &D_0600BEAC,
+    };
+
+    static u8 D_809440C8[] = {
+        0, 0, 0, 0, 2, 0, 0, 2, 2,
+    };
+
+    this->unk_284 = SkelAnime_GetFrameCount(&D_809440A4[arg1]->common);
+    SkelAnime_ChangeAnim(&this->skelAnime, D_809440A4[arg1], 1.0f, 0.0f, this->unk_284, D_809440C8[arg1], -4.0f);
+}
+
+void func_809437C8(EnDaiku* this) {
+    if ((this->unk_288 != -1) && (this->unk_258 != 0)) {
+        if (!func_8013D68C(this->unk_258, this->unk_25C, &this->unk_26C)) {
+            Actor_MarkForDeath(&this->actor);
+        }
+    }
+}
+
+void func_80943820(EnDaiku* this) {
+    s32 day = gSaveContext.day - 1;
+
+    switch (this->unk_278) {
+        case 0:
+        case 1:
+            func_8094373C(this, 0);
+            break;
+
+        case 2:
+            func_8094373C(this, 7);
+            break;
+
+        case 3:
+            func_8094373C(this, 6);
+            break;
+    }
+
+    this->unk_28C = (day * 4) + this->unk_278;
+    this->actor.textId = sTextIds[this->unk_28C];
+    this->unk_280 = this->actor.shape.rot.y;
+    this->unk_28A = 0;
+    this->actionFunc = func_809438F8;
+}
+
+void func_809438F8(EnDaiku* this, GlobalContext* globalCtx) {
+    f32 currentFrame = this->skelAnime.animCurrentFrame;
+    s32 pad;
+    s32 day = gSaveContext.day - 1;
+    s32 pad2;
```

Comment:
```suggestion
    s32 pad;
    s32 day = gSaveContext.day - 1;
    s32 pad2;
```
Is it possible to have the -1 used later? The temp variable day doesn't really represent the right day, it's off by one. Does that make sense?

Reply:
I can't get that to work, the - 1 is done beore all the function calls, trying to delay it until after won't move it back up unfortunately. The same -1 thing shows up in Daiku2 as well.